src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -1173,12 +1178,34 @@ public async Task MigrateDatabaseOffline()
                         }
                         catch (Exception e)
                         {
-                            result.AddError(e.ToString());
-                            throw;
+                            if (e is OperationCanceledException || e is ObjectDisposedException)
+                            {
+                                var killed = ProcessExtensions.TryKill(process);
+                                result.AddError($"Exception: {e}, process pid: {process.Id}, killed: {killed}");
+                                throw;
+                            }
+                            else
+                            {
+                                string errorString;
+                                try
+                                {
+                                    var processErrorString = await ProcessExtensions.ReadOutput(process.StandardError).ConfigureAwait(false);
+                                    errorString = $"Error occurred during migration. Process error: {processErrorString}, exception: {e}";
+                                }
+                                catch
+                                {
+                                    errorString = $"Error occurred during migration. Exception: {e}.";
+                                }
+                                result.AddError($"{errorString}");
+                                onProgress.Invoke(result.Progress);
+
+                                var killed = ProcessExtensions.TryKill(process);
+                                throw new InvalidOperationException($"{errorString} Process pid: {process.Id}, killed: {killed}");
+                            }
                         }
                         finally
                         {
-                            if (string.IsNullOrEmpty(tmpFile) == false)
+                            if (process.HasExited && string.IsNullOrEmpty(tmpFile) == false)
```

Comment:
Are we expecting here that it might not be killed/exited earlier?

Reply:
If it didn't, we need to kill it, and the code above certain suggest we try

Reply:
done